packages/upload-client/test/unixfs.test.js

```diff
@@ -99,12 +99,48 @@ describe('UnixFS', () => {
         new File(['a file, not a directory'], 'file.txt'),
         new File(['a file in a file!!!'], 'file.txt/another.txt'),
       ]),
-      { message: '"file.txt" cannot be a file and a directory' }
+      { message: '"file.txt/another.txt" cannot be a file and a directory' }
     ))
 
   it('configured to use raw leaves', async () => {
     const file = new Blob(['test'])
     const { cid } = await encodeFile(file)
     assert.equal(cid.code, raw.code)
   })
+
+  it('callback for each directory entry link', async () => {
+    const files = [
+      new File(['file'], 'file.txt'),
+      new File(['another'], '/dir/another.txt'),
+    ]
+    /** @type {import('../src/types.js').DirectoryEntryLink[]} */
+    const links = []
+    await encodeDirectory(files, { onDirectoryEntryLink: (l) => links.push(l) })
```

Comment:
hmm, but the callback pushes onto `links` and then we assert on the values...if it wasn't invoked the test would fail as it is...am I missing something?

Reply:
that's right! 🤦🏼 indentation looked to inside callback but you right